src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -43,18 +72,29 @@ def get_source_schema(cls):
         source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file."
         return source_schema
 
-    def __init__(self, file_path: FilePathType, stub_test: Optional[bool] = False):
-        super().__init__(file_path=str(file_path))
+    def __init__(
+        self,
+        file_path: FilePathType,
+        stub_test: Optional[bool] = False,
+        spikeextractors_backend: Optional[bool] = False,
+    ):
+
+        if spikeextractors_backend:
+            self.RX = se.SpikeGLXRecordingExtractor
+            super().__init__(file_path=str(file_path))
+            self.recording_extractor = OldToNewRecording(oldapi_recording_extractor=self.recording_extractor)
+        else:
+            file_path = Path(file_path)
+            folder_path = file_path.parent
+            stream_id = "".join(file_path.suffixes[:-1])[1:]
+            super().__init__(folder_path=folder_path, stream_id=stream_id)
+            self.source_data["file_path"] = str(file_path)
+
         if stub_test:
             self.subset_channels = [0, 1]
```

Comment:
I'd prefer to remove this bit that subsets channels under stub mode; it's not expensive to include a multiple of 384 channels compared to a 20-30kHz temporal resolution, so might as well just stub time for all channels. This helps a lot when testing electrode metadata as well as the overall writing process